src/linked_hash_map.rs

```diff
@@ -1273,10 +1231,11 @@ impl<'a, K, V, S> RawVacantEntryMut<'a, K, V, S> {
             new_node.as_mut().put_entry((key, value));
             attach_before(new_node, NonNull::new_unchecked(self.values.as_ptr()));
 
-            let node = *self
+            let node = self
                 .entry
-                .insert_with_hasher(hash, new_node, (), move |k| hasher((*k).as_ref().key_ref()))
-                .0;
+                .into_table()
+                .insert_unique(hash, new_node, move |k| hasher((*k).as_ref().key_ref()))
```

Comment:
One thing I missed when reviewing this is the behavior of `RawVacantEntryMut`, which has subtly changed (but I think it is still acceptable behavior).

If you produce a `RawVacantEntryMut` for a specific key, then insert a totally unrelated key using the `RawVacantEntryMut` you receive, you can end up with duplicate keys (which is not a safety issue, only a correctness one). This seems to match the same behavior in hashbrown, so this is fine, but I should have brought this up during review to ask to make sure I understood everything correctly.

Reply:
I don't see what subtle change you mean. Yes, you can do incorrect things here, but nothing *new* -- AFAICT it ends up doing the exact same insertion on the inner `RawTable`, before and after my change. Am I missing something?

Before: `RawVacantEntryMut::insert_with_hasher` -> `RawTable::insert_entry` -> `RawTable::insert`
After: `HashTable::insert_unique` -> `RawTable::insert`

Reply:
Oh no you're exactly right, it's exactly the same as it was before, I just didn't realize what the previous behavior was.